### DISCUSSION TITLE
Question about other IDEs

### DISCUSSION THREAD

**Discussion by BlakeTNC** [revelc/formatter-maven-plugin#551] [Q&A]
Hi, we have a multi-IDE project. 
Does this plugin work when building under other IDEs besides Eclipse, like Netbeans or IntelliJ? 
(Or does it only work for an Eclipse user?)
Will it produce identical formatting under the other IDEs?

Thank you.

**Comment by hazendaz** [comment]
The plugin is entirely self contained, it does not matter what IDE is used or if no IDE at all.  Its driven entirely by maven and configuration provided to that.  It happens to use Eclipse to do the formatting for java/javascript.  It uses jsoup for html/xhtml.  It uses cssparser for css.  And a separately exposed sub project for the javascript portion as well as separately exposed project to format xml, which are both under revelc project.

As to identical, you would have to configure whatever IDE or editor (ie notepad++) to use same formatting.  Most product defaults do not come with very modern formatting setup as most try to keep legacy alive.  I would suggest using google rules for styling which checkstyle also implements and/or any further custom rules you need to get your baseline for formatting across platforms.

**Comment by ctubbsii** [comment]
I would say that the entire point of this plugin is to make the formatting standard, regardless of what IDE or formatting conventions each individual developer/contributor uses. The plugin allows you to standardize the formatting for your project during the Maven build, so that individual IDE and user conventions don't matter anymore. This plugin applies Eclipse formatting code and configuration to the project, but you don't have to use Eclipse. You can use whatever IDE and conventions you want, and then just run this plugin to standardize your project's formatting.
